server/engine/src/main/java/org/eclipse/lsp/cobol/core/visitor/VisitorHelper.java

```diff
@@ -115,22 +117,18 @@ public static List<String> retrievePicTexts(List<org.eclipse.lsp.cobol.core.Cobo
    * @return the list of value intervals
    */
   public static List<ValueInterval> retrieveValueIntervals(List<org.eclipse.lsp.cobol.core.CobolParser.DataValueIntervalContext> contexts) {
-    return contexts.stream()
-            .map(
-                    context ->
-                            new ValueInterval(
-                                    context.dataValueIntervalFrom().getText(),
-                                    ofNullable(context.dataValueIntervalTo())
-                                            .map(org.eclipse.lsp.cobol.core.CobolParser.DataValueIntervalToContext::literal)
-                                            .map(ParserRuleContext::getText)
-                                            .map(String::toUpperCase)
-                                            .orElse(null),
-                                    ofNullable(context.dataValueIntervalTo())
-                                            .map(org.eclipse.lsp.cobol.core.CobolParser.DataValueIntervalToContext::thruToken)
-                                            .map(ParserRuleContext::getText)
-                                            .map(String::toUpperCase)
-                                            .orElse(null)))
-            .collect(toList());
+    List<ValueInterval> list = new ArrayList<>();
+    for (CobolParser.DataValueIntervalContext context: contexts) {
+      String from = context.dataValueIntervalFrom().getText();
```

Comment:
This and the grammar for `dataValueInterval` might be worth looking into (in a separate PR) wrt toUpperCase operation.